src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -437,51 +436,57 @@ def invert(self, ore_pol):
             sage: phi_r5 = cat.random_object(5)
             sage: phi_r5.invert(phi_r5(a)) == a
             True
+
+        ::
+
+            sage: B.<X> = Fq[]
+            sage: phi_r5.invert(X)
+            Traceback (most recent call last):
+            ...
+            TypeError: input must be an Ore polynomial
+
         """
         deg = ore_pol.degree()
         r = self.rank()
-        base_over_Fq = self.base_over_constants_field()
-        if ore_pol not in self._ore_polring:
+        E = self.base()
+        K = self.base_over_constants_field()
+        if ore_pol in self._ore_polring:
+            ore_pol = self._ore_polring(ore_pol)
+        else:
             raise TypeError('input must be an Ore polynomial')
-        if ore_pol.degree() == 0:
-            coord = base_over_Fq(self._base(ore_pol)).vector()
-            if coord.nonzero_positions == [0]:
-                return self._Fq(coord[0])
-        if ore_pol == 0:
-            return self._Fq.zero()
-        if deg % r != 0:
-            raise ValueError('input must be in the image of the Drinfeld '
-                             'module')
 
+        if deg <= 0:
```

Comment:
Add a small comment:

```suggestion
        # Trivial cases
        if deg <= 0:
```